test/service.go

```diff
@@ -65,9 +67,53 @@ func CreateService(ctx *Context, name, namespace, image string) (*servingv1beta1
 	return service, nil
 }
 
+func WaitForControllerEnvironment(ctx *Context, ns, envName, envValue string) error {
+	return wait.PollImmediate(Interval, 10*time.Minute, func() (bool, error) {
+		pods, err := ctx.Clients.Kube.CoreV1().Pods(ns).List(metav1.ListOptions{
+			LabelSelector: "app=controller",
+		})
+		if apierrs.IsUnauthorized(err) {
+			// These errors happen when resetting the proxy value. Just retry.
+			return false, nil
+		} else if err != nil {
+			return false, err
+		}
+		for _, pod := range pods.Items {
+			if !isPodReady(pod) {
+				return false, nil
+			}
+			for _, container := range pod.Spec.Containers {
+				for _, e := range container.Env {
+					if e.Name == envName && e.Value != envValue {
+						return false, nil
+					}
+				}
+
+			}
+		}
+		// We only reach here if all controller pods are ready and have the
```

Comment:
Ugh, this is of course wrong. You might want to tune this back to the implementation you had before. Sorry I screwed this one up.

Reply:
No issues i will change
I m just verifying by testing locally now